pkg/plugins/policies/meshhealthcheck/api/v1alpha1/validator.go

```diff
@@ -28,17 +28,20 @@ func validateTop(targetRef common_api.TargetRef) validators.ValidationError {
 	return targetRefErr
 }
 
-func validateTo(to []To) validators.ValidationError {
+func validateTo(topTargetRef common_api.TargetRef, to []To) validators.ValidationError {
 	var verr validators.ValidationError
 	for idx, toItem := range to {
 		path := validators.RootedAt("to").Index(idx)
 		verr.AddErrorAt(path.Field("targetRef"), mesh.ValidateTargetRef(toItem.TargetRef, &mesh.ValidateTargetRefOpts{
 			SupportedKinds: []common_api.TargetRefKind{
 				common_api.Mesh,
 				common_api.MeshService,
+				common_api.MeshExternalService,
 			},
 		}))
-
+		if toItem.TargetRef.Kind == common_api.MeshExternalService && topTargetRef.Kind != common_api.Mesh {
+			verr.AddViolationAt(path.Field("targetRef.kind"), "kind MeshExternalService is only allowed with targetRef.kind: Mesh")
```

Comment:
I think it worth mentioning something like `as it's configured on the Zone Egress and shared by all clients in the mesh`